### PR TITLE
3533 Update PyTorch docker to 21.12

### LINKED ISSUE #3533: Update base PyTorch docker 21.12
**Is your feature request related to a problem? Please describe.**
@yiheng-wang-nv and I are testing a performance drop issue in the spleen segmentation task when comparing MONAI 0.5 and 0.8. And I found that NVIDIA just released 21.12 PyTorch docker:
https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch
Will update it in MONAI docker and test.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3533 .

### Description
This PR updated the PyTorch docker to 21.12.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
I tested the `test_integration_segmentation_3d.py`, `test_integration_classification_2d.py`, `test_integration_workflows.py`, `test_integration_fast_train.py` locally on V100 GPU, all passed successfully.

Thanks.

File: .github/workflows/pythonapp-gpu.yml [hunk 1]
```diff
@@ -43,9 +43,9 @@ jobs:
           - environment: PT19+CUDA114
             # we explicitly set pytorch to -h to avoid pip install error
             # https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes
-            # 21.10: 1.10.0a0+0aef44c
+            # 21.12: 1.11.0a0+b6df043
```
Comment (wyli) [inline .github/workflows/pythonapp-gpu.yml:46]: Could you update the environment string to PT110 as it's after v1.10

Comment (Nic-Ma) [reply .github/workflows/pythonapp-gpu.yml:46]: Hi @wyli ,

Thanks for your tips, updated PT110 and CUDA version in the latest commit.
```
# nvidia-smi
Wed Dec 22 06:31:47 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.44       Driver Version: 440.44       CUDA Version: 11.5     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-PCIE...  Off  | 00000000:02:00.0 Off |                    0 |
| N/A   35C    P0    35W / 250W |      0MiB / 32510MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla V100-PCIE...  Off  | 00000000:03:00.0 Off |                    0 |
| N/A   35C    P0    37W / 250W |      0MiB / 32510MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

Thanks.

File: .github/workflows/pythonapp-gpu.yml [hunk 2]
```diff
@@ -23,7 +23,7 @@ jobs:
           - "PT17+CUDA110"
           - "PT18+CUDA102"
```
Comment (wyli) [inline .github/workflows/pythonapp-gpu.yml:24]: I forgot to mention that now there is no test for v1.9 pytorch, please replace at least one of the previous tests with a 1.9 test

Comment (Nic-Ma) [reply .github/workflows/pythonapp-gpu.yml:24]: OK, sure, let me update it.

Thanks for your reminder.

Comment (Nic-Ma) [reply .github/workflows/pythonapp-gpu.yml:24]: I added the PT19 test back and deleted the previous PT18 test with `21.02` docker, I feel maybe it's typo in the previous config:
https://github.com/Project-MONAI/MONAI/blob/dev/.github/workflows/pythonapp-gpu.yml#L42
If not typo, we also covered `21.02` in the crown test.

Thanks.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me. I'll update blossom for the same.

**Pull request merged**